src/array/cpu/csr_linegraph.cc

```diff
@@ -0,0 +1,60 @@
+/*!
+ *  Copyright (c) 2020 by Contributors
+ * \file array/cpu/coo_line_graph.cc
+ * \brief CSR LineGraph
+ */
+
+#include <dgl/array.h>
+#include <numeric>
+#include <algorithm>
+#include <vector>
+#include <iterator>
+
+namespace dgl {
+namespace aten {
+namespace impl {
+
+template <DLDeviceType XPU, typename IdType>
+CSRMatrix CSRLineGraph(const CSRMatrix &csr, bool backtracking) {
+  const int64_t nnz = csr.indices->shape[0];
+  const IdType *indptr_data = static_cast<IdType *>(csr.indptr->data);
+  const IdType *indices_data = static_cast<IdType *>(csr.indices->data);
+  std::vector<IdType> new_indptr = {0};
```

Comment:
reserve space of size `nnz + 1`

Reply:
If backtracing is False, The space is not nnz+1.